lib/rucio/client/uploadclient.py

```diff
@@ -312,7 +312,7 @@ def upload(self, sources_with_settings, summary_file_path=None):
                 else:
                     logger.error('Failed to upload file %s' % basename)
                     # TODO trace?
-                    continue  # skip attach_did and update_states for this file
+                    raise RucioException('Failed to upload file %s' % basename)
```

Comment:
it might be worth to add a new exception in `lib/rucio/common/exception.py` like FailedUpload and raises it there.

Reply:
Yes, this should be an exception. I am actually surprised that we didn't have one before 😄